gensim/models/phrases.py

```diff
@@ -165,7 +165,10 @@ def __str__(self):
     def learn_vocab(self, sentences):
         """Collect unigram/bigram counts from the `sentences` iterable."""
         if not self.recode_to_utf8 and sentences is not None:
-            sentence = list(next(iter(sentences)))
+            try:
+                sentence = list(next(iter(sentences)))
+            except:
+                raise ValueError("Input can not be empty list or generator.")
```

Comment:
@piskvorky  this concern was raised by @jayantj in [review comment above](https://github.com/RaRe-Technologies/gensim/pull/1413#discussion_r125001735)